mm2src/lp_ordermatch.rs

```diff
@@ -4351,6 +4386,17 @@ pub async fn create_maker_order(ctx: &MmArc, req: SetPriceReq) -> Result<MakerOr
         let mut my_maker_orders = ordermatch_ctx.my_maker_orders.lock();
         my_maker_orders.insert(new_order.uuid, Arc::new(AsyncMutex::new(new_order.clone())));
     }
+
+    let balance = match base_coin.my_spendable_balance().compat().await {
+        Ok(balance) => Some(balance),
+        Err(_) => None,
+    };
+
+    let ctx_weak = ctx.weak();
+    let ticker = base_coin.ticker().to_string();
+    let uuid = new_order.uuid;
+    spawn(async move { check_balance_update_loop(ctx_weak, ticker, balance, uuid).await });
```

Comment:
It now spawns multiple loops for the same coin, if it’s used as base for different orders. It should run only a single background task per coin despite the number of orders that are using it. But if all these orders are canceled, the loop should stop.

Reply:
There's a check that stops the loop if the order of the coin does not exist anymore, so even if multiple loops are started for the same coin only one will remain in 30 seconds. But still I can prevent creating new loops for the same coin by checking if there is already an order for the coin.

Reply:
No you're right, currently it's possible to create multiple loops for the same base coin. But the additional check for orders at loop creation time will fix it.